pennylane/spin/spin_hamiltonian.py

```diff
@@ -93,4 +96,187 @@ def transverse_ising(
+       neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
+                    Default is 1 (nearest neighbour).
+
+
+    Returns:
+       pennylane.LinearCombination: Hamiltonian for the heisenberg model.
+
+    **Example**
+
+    >>> n_cells = [2,2]
+    >>> J = [[0.5, 0.5, 0.5]]
+    >>> spin_ham = heisenberg("Square", n_cells, coupling=J)
+    >>> print(spin_ham)
+    0.5 * (X(0) @ X(1)) + 0.5 * (Y(0) @ Y(1)) + 0.5 * (Z(0) @ Z(1)) + 0.5 * (X(0) @ X(2)) +
+    0.5 * (Y(0) @ Y(2)) + 0.5 * (Z(0) @ Z(2)) + 0.5 * (X(1) @ X(3)) + 0.5 * (Y(1) @ Y(3)) +
+    0.5 * (Z(1) @ Z(3)) + 0.5 * (X(2) @ X(3)) + 0.5 * (Y(2) @ Y(3)) + 0.5 * (Z(2) @ Z(3))
+    """
+
+    lattice = _generate_lattice(lattice, n_cells, boundary_condition, neighbour_order)
+
+    if coupling is None:
+        coupling = [[1.0, 1.0, 1.0]]
+    coupling = math.asarray(coupling)
+    if coupling.ndim == 1:
+        coupling = math.asarray([coupling])
+
+    if coupling.shape not in [(neighbour_order, 3), (3, lattice.n_sites, lattice.n_sites)]:
+        raise ValueError(
+            f"Coupling shape should be equal to {neighbour_order}x3 or 3x{lattice.n_sites}x{lattice.n_sites}"
+        )
+
+    hamiltonian = 0.0 * qml.I(0)
+    if coupling.shape == (neighbour_order, 3):
+        for edge in lattice.edges:
+            i, j, order = edge[0], edge[1], edge[2]
+            hamiltonian += (
+                coupling[order][0] * (X(i) @ X(j))
+                + coupling[order][1] * (Y(i) @ Y(j))
+                + coupling[order][2] * (Z(i) @ Z(j))
+            )
+    else:
+        for edge in lattice.edges:
+            i, j = edge[0], edge[1]
+            hamiltonian += (
+                coupling[0][i][j] * X(i) @ X(j)
+                + coupling[1][i][j] * Y(i) @ Y(j)
+                + coupling[2][i][j] * Z(i) @ Z(j)
+            )
+
+    return hamiltonian.simplify()
+
+
+def fermihubbard(
+    lattice,
+    n_cells,
+    hopping=None,
+    interaction=1.0,
+    boundary_condition=False,
+    neighbour_order=1,
+    mapping="jordan_wigner",
+):
+    r"""Generates the Hubbard model on a lattice.
+
+    The Hamiltonian is represented as:
+
+    .. math::
+
+        \hat{H} = -t\sum_{<i,j>, \sigma}(c_{i\sigma}^{\dagger}c_{j\sigma}) + U\sum_{i}n_{i \uparrow} n_{i\downarrow}
+
+    where ``t`` is the hopping term representing the kinetic energy of electrons, and ``U`` is the on-site Coulomb interaction,
+    representing the repulsion between electrons.
+    This function assumes there are two fermions with opposite spins on each site.
+
+    Args:
+       lattice (str): Shape of the lattice. Input Values can be ``'chain'``, ``'square'``,
+                      ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+       n_cells (List[int]): Number of cells in each direction of the grid.
+       hopping (float or List[float] or List[math.array(float)]):Hopping interaction between spins, it can be a
```

Comment:
"between spins" sounds a bit odd here.

Reply:
how about "strength of hopping between neighboring sites."